Add testing instructions to README.md (#1894)
Automattic/themes: README.md

* Reference any existing issues that relate to your PR. This allows everyone to easily see all related discussions.

By contributing code to our free themes, you grant its use under the [GNU General Public License v2 (or later)](LICENSE).

## Testing pull requests

### Using GitHub 

1. Clone repository locally `git clone git@github.com:Automattic/themes.git`

> 2. Look up featured branch of pull request, e.g. `update/#1889` 

Reviewer: ```suggestion
2. [Identify the branch](https://cloudup.com/cAy1j0D8tvH) of the pull request, e.g. `update/#1889` 
```

^ Links to to an image showing where to find this.

3. Check out featured branch of pull request, e.g. `git checkout update/#1889`
4. Symlink or copy affected theme OR zip affected theme and import into WordPress site

### Manually download

> 1. Look up featured branch of pull request, e.g. `update/#1889` 

Reviewer: ```suggestion
1. [Click on the branch](https://cloudup.com/cAy1j0D8tvH) of the pull request, e.g. `update/#1889` 
```

> 2. Download featured branch of pull request, e.g. `https://github.com/Automattic/themes/archive/update/%231889.zip`

Reviewer: ```suggestion
2. Change the `/tree/` part of the branch's URL to `archive`, and add a `.zip` to the end to download a zip of the branch. e.g. `https://github.com/Automattic/themes/tree/update/%231889` would become `https://github.com/Automattic/themes/archive/update/%231889.zip`
```

^ Just makes things a little more clear.
